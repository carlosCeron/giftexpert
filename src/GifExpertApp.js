import React, { Fragment, useState } from 'react'
import CategoryAdd from './components/CategoryAdd';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({categoriesDef = []}) => {

  const [categorias, setCategorias] = useState(categoriesDef);

  return (
    <Fragment>
      <h2>GifExpertApp</h2>
      <CategoryAdd  setCategorias={setCategorias} />
      <hr/>
      {categorias.map(categoria => <GifGrid key={categoria} category={categoria} />)}
    </Fragment>
  )
}

export default GifExpertApp
