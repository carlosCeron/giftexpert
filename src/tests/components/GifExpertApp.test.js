import React from 'react';
import { shallow } from "enzyme";
import GifExpertApp from "../../GifExpertApp";


describe('testing GifExpertApp component', () => {


  test('should same snapshot with the component markup', () => {
    
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();

  });

  test('should show a category list', () => {
    const categorias = ['one', 'two'];
    const wrapper = shallow(<GifExpertApp categoriesDef={categorias} />);
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toEqual(categorias.length);
  })
  


});