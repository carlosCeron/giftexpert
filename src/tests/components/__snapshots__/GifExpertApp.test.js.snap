// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing GifExpertApp component should same snapshot with the component markup 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setCategorias={[Function]}
  />
  <hr />
</Fragment>
`;

exports[`testing GifExpertApp component should show a category list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setCategorias={[Function]}
  />
  <hr />
  <GifGrid
    category="one"
    key="one"
  />
  <GifGrid
    category="two"
    key="two"
  />
</Fragment>
`;
